Purchases: Add tracks events for cancellation survey

markryall opened the pull request:
The cancellation survey does not currently fire any analytics events so it is not possible to see how this survey is being used unless it is completely filled in and submitted.

This change adds the following tracks events:

* `calypso_purchases_cancel_form_start` when the 'remove <product>' button is clicked
* `calypso_purchases_cancel_form_select_radio_option` when the radio button options are selected
* `calypso_purchases_cancel_form_step` when the 'next' and 'previous' buttons are clicked
* `calypso_purchases_cancel_form_close` when the survey is cancelled
* `calypso_purchases_cancel_form_submit` when the survey is submitted (and plan is cancelled)

# To test

There are two completely distinct sets of components for cancelling/removing a plan depending on whether a refund is due.

You will need a user with a site associated with a paid plan and another site with a paid plan due for a refund. In chrome, show the network tab in developer tools with a filter for `t.gif` requests.

* browse to `/me/purchases`
* click on the plan
* click on 'Remove <product>'
* fill in survey, click previous, next, cancel, etc. and you should see each of the above events firing.

alisterscott commented:
Tried removing Premium added via SA and it doesn't nothing, console error:

`Uncaught TypeError: Cannot read property 'productSlug' of undefined`

markryall commented:
Ah yes - thanks for picking that up - the property is called `purchase` in one component and `selectedPurchase` in the other.  I've changed it and added lodash `get` for safety.

alisterscott commented:
Tested all the events for a SA added plan - which is the cancel flow - all good
I couldn't see the events when trying to refund a recently added plan for an e2e test - not sure why
